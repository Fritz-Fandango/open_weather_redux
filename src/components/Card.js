import React, { Component } from 'react';
import '../styles/Card.scss';


function formatDate (timestamp) {
    const event = new Date(timestamp);

    return event.toLocaleDateString('en-US');
}

function formatTime (timestamp) {
    const event = new Date(timestamp);

    return event.toLocaleTimeString('en-US');
}

class Card extends Component {
    render() {
        return (
            <div className='card'>
                <img
                    src={`http://openweathermap.org/img/wn/${this.props.icon}@2x.png`}
                    alt='Avatar'
                    style={{width: '100%' }}
                />
                <div className='container'>
                    <p>Date: {formatDate(this.props.dateTime)}</p>
                    <p>Min temp: {this.props.minTemp}&deg;F</p>
                    <p>Max temp: {this.props.maxTemp}&deg;F</p>
                    <p>Humidty: {this.props.humidity}%</p>
                    <p>Sunrise: {formatTime(this.props.sunrise)}</p>
                    <p>Sunset: {formatTime(this.props.sunset)}</p>
                </div>
            </div>
        )
    }
}

export default Card;
