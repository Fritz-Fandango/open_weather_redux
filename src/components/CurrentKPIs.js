import React, { Component } from 'react';
import '../styles/CurrentKPIs.scss';

function formatTime (timestamp) {
    const event = new Date(timestamp);

    return event.toLocaleTimeString('en-US');
}

class CurrentKPIs extends Component {
    render() {
        return (
            <section className="kpiHeader">
                <h2>Today's weather</h2>
                <hr />
                <div className="cardDeck">
                    <button className="dataCard">
                        <h3>{formatTime(this.props.current.dt)}</h3>
                        <h4>Time Stamp</h4>
                        <p>Nullam volutpat et eros ut tristique.</p>
                    </button>
                    <button className="dataCard">
                        <h3>{this.props.current.temp} &deg;F</h3>
                        <h4>Current Temperature</h4>
                        <p>Nullam volutpat et eros ut tristique.</p>
                    </button>
                    <button className="dataCard">
                        <h3>{this.props.current.feels_like} &deg;F</h3>
                        <h4>Feels Like</h4>
                        <p>Nullam volutpat et eros ut tristique.</p>
                    </button>
                    <button className="dataCard">
                        <h3>{this.props.current.pressure} hPa</h3>
                        <h4>Atmospheric Pressure</h4>
                        <p>Nullam volutpat et eros ut tristique.</p>
                    </button>
                    <button className="dataCard">
                        <h3>{this.props.current.wind_speed} mi/hr</h3>
                        <h4>Current wind speed</h4>
                        <p>Nullam volutpat et eros ut tristique.</p>
                    </button>
                </div>
            </section>
        )
    }
};

export default CurrentKPIs;
