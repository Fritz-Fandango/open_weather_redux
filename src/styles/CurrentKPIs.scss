@import url('https://fonts.googleapis.com/css2?family=Open+Sans:wght@600;700&display=swap');

.kpiHeader {
  background: #E7F3F1;
  font-family: 'Open Sans', sans-serif;
  padding: 0;
  width: 100vw;

  h2 {
    font-family: 'Open Sans', sans-serif;
  }
  .cardDeck {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-evenly;

    .dataCard {
      overflow: hidden;
      border-radius: .5em;
      text-decoration: none;
      background: white;
      margin: 1em;
      padding: 2.75em 2.5em;
      box-shadow: 0 1.5em 2.5em -.5em rgba(#000000, .1);
      transition: transform .45s ease, background .45s ease;

      h3 {
        color: #2E3C40;
        font-size: 3.0em;
        font-weight: 600;
        line-height: 1;
        padding-bottom: .5em;
        margin: 0;
        border-bottom: 2px solid #753BBD;
        transition: color .45s ease, border .45s ease;
      }

      h4 {
        color: #627084;
        text-transform: uppercase;
        font-size: 1.125em;
        font-weight: 700;
        line-height: 1;
        letter-spacing: 0.1em;
        transition: color .45s ease;
      }

      p {
        opacity: 0;
        color: #FFFFFF;
        font-weight: 600;
        line-height: 1.8;
        margin: 0 0 1.25em;
        transform: translateY(-1em);
        transition: opacity .45s ease, transform .5s ease;
      }

      &:hover {
        background: #753BBD;
        transform: scale(1.02);

        h3 {
          color: #FFFFFF;
          border-bottom-color: #A754C4;
        }

        h4 {
          color: #FFFFFF;
        }

        p {
          opacity: 1;
          transform: none;
        }

        .link-text {
          color: #FFFFFF;

          svg {
            animation: point 1.25s infinite alternate;

            path {
              fill: #FFFFFF;
            }
          }
        }
      }
    }
  }


  @keyframes point {
    0% {
     transform: translateX(0);
    }
    100% {
      transform: translateX(.125em);
    }
  }
}
