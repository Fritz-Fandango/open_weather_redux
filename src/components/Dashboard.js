import React, { Component } from 'react';
import { connect } from 'react-redux';
import { weatherFetchData } from '../actions/index.js';

// Components
import CurrentKPIs from './CurrentKPIs';
import Card from './Card';

// Styles
import '../styles/Dashboard.scss';

class Dashboard extends Component {
    componentDidMount() {
        this.props.fetchData();
    }

    render() {
        if (this.props.hasErrored) {
            return <p>Sorry! There was an error loading the weather</p>;
        }

        if (this.props.isLoading) {
            return <p>Loadingâ€¦</p>;
        }

        return (
            <div className='dashboard'>
                <h1>Weather App</h1>
                {this.props.weather.current &&
                    <CurrentKPIs current={this.props.weather.current} />
                }
                <div className='title'>
                    <h2>Forecasted weather</h2>
                    <hr />
                </div>
                {this.props.weather?.daily?.map(today =>
                    <Card
                        key={today.dt}
                        dateTime={today.dt}
                        humidity={today.humidity}
                        icon={today.weather[0].icon}
                        minTemp={today.temp.min}
                        maxTemp={today.temp.max}
                        sunrise={today.sunrise}
                        sunset={today.sunset}
                    />
                )}
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        weather: state.weather,
        hasErrored: state.weatherHasErrored,
        isLoading: state.weatherIsLoading
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        fetchData: () => dispatch(weatherFetchData())
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
