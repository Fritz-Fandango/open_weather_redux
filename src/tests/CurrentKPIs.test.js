import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import renderer from 'react-test-renderer';
import { formatTime } from '../components/Card';

import CurrentKPIs from '../components/CurrentKPIs';

configure({adapter: new Adapter()});

describe('Testing date formatting', () => {
    const timestamp = 1618308000; // 1/19/1970 10:31:48 AM

    it('should return a valid time string', () => {
        const formattedDate = formatTime(timestamp);

        expect(formattedDate).toBe('10:31:48 AM');
    });
});

describe('Testing CurrentKPIs component', () => {
    const propsObj = {
        dt: 1618308000,
        temp: 275.09,
        feels_like: 279.79,
        pressure: 15,
        wind_speed: 300
    };

    it('renders CurrentKPIs', () => {
        const wrapper = <CurrentKPIs current={propsObj} />;

        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
