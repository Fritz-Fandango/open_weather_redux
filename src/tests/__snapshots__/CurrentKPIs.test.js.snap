// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CurrentKPIs component renders CurrentKPIs 1`] = `
<section
  className="kpiHeader"
>
  <h2>
    Today's weather
  </h2>
  <hr />
  <div
    className="cardDeck"
  >
    <button
      className="dataCard"
    >
      <h3>
        10:31:48 AM
      </h3>
      <h4>
        Time Stamp
      </h4>
      <p>
        Nullam volutpat et eros ut tristique.
      </p>
    </button>
    <button
      className="dataCard"
    >
      <h3>
        275.09
         °F
      </h3>
      <h4>
        Current Temperature
      </h4>
      <p>
        Nullam volutpat et eros ut tristique.
      </p>
    </button>
    <button
      className="dataCard"
    >
      <h3>
        279.79
         °F
      </h3>
      <h4>
        Feels Like
      </h4>
      <p>
        Nullam volutpat et eros ut tristique.
      </p>
    </button>
    <button
      className="dataCard"
    >
      <h3>
        15
         hPa
      </h3>
      <h4>
        Atmospheric Pressure
      </h4>
      <p>
        Nullam volutpat et eros ut tristique.
      </p>
    </button>
    <button
      className="dataCard"
    >
      <h3>
        300
         mi/hr
      </h3>
      <h4>
        Current wind speed
      </h4>
      <p>
        Nullam volutpat et eros ut tristique.
      </p>
    </button>
  </div>
</section>
`;
